hands_on/checkout_commits.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+from exercise_utils.cli import run_command
+from exercise_utils.gitmastery import create_start_tag
+from exercise_utils.git import init, add, commit, tag
+from pathlib import Path
+
+__requires_git__ = True
+__requires_github__ = False
+
+def download(verbose: bool):
+    """
+    hp-checkout-commits (T4L4)
+    - Creates a local repo 'things' with multiple commits and tags.
+    - Students can practice checking out commits by hash, tag, or HEAD~N references.
+    """
+    
+    target_dir = Path("things")
+
+    # Clean existing sandbox if re-downloading
+    if target_dir.exists():
+        import shutil
+        shutil.rmtree(target_dir)
```

Comment:
This already happens via the app: https://git-mastery.github.io/developers/docs/architecture/download-workflow/#hands-on-download

So this check can be omitted